MoreLinq/NestedLoops.cs

```diff
@@ -41,21 +41,18 @@ public static IEnumerable<Action> NestedLoops(this Action action, IEnumerable<in
             if (action == null) throw new ArgumentNullException(nameof(action));
             if (loopCounts == null) throw new ArgumentNullException(nameof(loopCounts));
 
-            using (var iter = loopCounts.GetEnumerator())
+            return _(); IEnumerable<Action> _()
             {
-                var loopCount = NextLoopCount(iter);
-                if (loopCount == null)
-                    return Enumerable.Empty<Action>(); // null loop
-                var loop = Enumerable.Repeat(action, loopCount.Value);
-                while ((loopCount = NextLoopCount(iter)) != null)
-                    loop = loop.Repeat(loopCount.Value);
-                return loop;
-            }
+                var count = loopCounts.Assert(n => n >= 0,
+                                              n => new ArgumentException("All loop counts must be greater than or equal to zero.", nameof(loopCounts)))
```

Comment:
`loopCounts` is an argument of `NestedLoops`. It cannot be raised during iteration after the call of `NestedLoops`. Behaviour may be left as undefined for negative counts or the exception raised must not derive from `ArgumentException`; you could use `Exception` or `InvalidOperationException`.